centreon/src/Core/TimePeriod/Infrastructure/Repository/DbReadTimePeriodRepository.php

```diff
@@ -0,0 +1,316 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\TimePeriod\Infrastructure\Repository;
+
+use Assert\AssertionFailedException;
+use Centreon\Domain\Log\LoggerTrait;
+use Centreon\Domain\RequestParameters\Interfaces\RequestParametersInterface;
+use Centreon\Domain\RequestParameters\RequestParameters;
+use Centreon\Infrastructure\DatabaseConnection;
+use Centreon\Infrastructure\RequestParameters\SqlRequestParametersTranslator;
+use Core\Common\Infrastructure\Repository\AbstractRepositoryRDB;
+use Core\TimePeriod\Application\Repository\ReadTimePeriodRepositoryInterface;
+use Core\TimePeriod\Domain\Exception\TimeRangeException;
+use Core\TimePeriod\Domain\Model\{Day, Template, ExtraTimePeriod, TimePeriod, TimeRange};
+
+class DbReadTimePeriodRepository extends AbstractRepositoryRDB implements ReadTimePeriodRepositoryInterface
+{
+    use LoggerTrait;
+
+    private SqlRequestParametersTranslator $sqlRequestTranslator;
+
+    /**
+     * @param DatabaseConnection $db
+     * @param SqlRequestParametersTranslator $sqlRequestTranslator
+     */
+    public function __construct(DatabaseConnection $db, SqlRequestParametersTranslator $sqlRequestTranslator)
+    {
+        $this->db = $db;
+        $this->sqlRequestTranslator = $sqlRequestTranslator;
+        $this->sqlRequestTranslator
+            ->getRequestParameters()
+            ->setConcordanceStrictMode(RequestParameters::CONCORDANCE_MODE_STRICT);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function exists(int $timePeriodId): bool
+    {
+        $this->info('Does the time period exist?', ['id' => $timePeriodId]);
+        $statement = $this->db->prepare(
+            $this->translateDbName('SELECT 1 FROM `:db`.timeperiod WHERE tp_id = :id')
+        );
+        $statement->bindValue(':id', $timePeriodId, \PDO::PARAM_INT);
+        $statement->execute();
+        return ! empty($statement->fetch());
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function findById(int $timePeriodId): ?TimePeriod
+    {
+        $this->info('Find time period by id', ['id' => $timePeriodId]);
+        $statement = $this->db->prepare(
+            $this->translateDbName('SELECT * FROM `:db`.timeperiod WHERE tp_id = :id')
+        );
+        $statement->bindValue(':id', $timePeriodId, \PDO::PARAM_INT);
+        $statement->execute();
+        if (($result = $statement->fetch(\PDO::FETCH_ASSOC)) !== false) {
+            /**
+             * @var array{
```

Comment:
Up to you, cf. idea of [@phpstan-type](https://phpstan.org/writing-php-code/phpdoc-types#local-type-aliases)